src/components/login/AuthDetails.tsx

```diff
@@ -0,0 +1,28 @@
+import { onAuthStateChanged, signOut } from "firebase/auth";
+import { useEffect, useState } from "react";
+import { auth } from '../../firebase-config';
+
+const AuthDetails = () => {
+    const [authUser, setAuthUser] = useState(null);
+
+    useEffect(() => {
+        const listen = onAuthStateChanged(auth, (user) => {
+            if (user) {
+                setAuthUser
+            } else {
+                setAuthUser(null);
+            }
+        })
+        return () => listen();
+    }, []);
+
+    const userSignOut = () => {
+        signOut(auth).then(() => {
+            console.log('signed out worked')
+        }).catch(error => console.log(error))
+    }
+
+    return <div>{authUser ? <><p>{`Signed In as ${authUser}`}</p><button onClick={userSignOut}> Sign Out</button> </> : <p>Signed Out</p>}</div> 
```

Comment:
Use an `authUser` property such as `email`. 
Something like this:
``{`Signed In as ${authUser.email}`}``

Reply:
I've tried doing it as you suggested, however I got an error. so instead I used this ${auth.currentUser?.email}

Reply:
oops! Yes, you're correct.